proto/proto.go

```diff
@@ -432,87 +432,87 @@ func CheckChunked(buf []byte) (chunkEnd int, full bool) {
 	return
 }
 
-// Message is an interface used to provide feedback or store dummy data for future use
-type Message interface {
+// Message is an interface used to provide protocol state or store dummy data for future use
+type ProtocolStateSetter interface {
```

Comment:
Since this interface has both setter(`SetProtocolState`) and getter(`ProtocolState`), it may be called `ProtocolState`